system/Debug/ExceptionHandler.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Debug;
+
+use CodeIgniter\Exceptions\PageNotFoundException;
+use Config\Paths;
+use Throwable;
+
+class ExceptionHandler extends BaseExceptionHandler
+{
+    /**
+     * Determines the correct way to display the error.
+     *
+     * @return mixed|void
+     */
+    public function handle()
+    {
+        if (! is_cli()) {
+            $this->response->setStatusCode($this->statusCode);
+            header(sprintf('HTTP/%s %s %s', $this->request->getProtocolVersion(), $this->response->getStatusCode(), $this->response->getReasonPhrase()), true, $this->statusCode);
+
+            // Display a JSON/XML response if we weren't asked for HTML
+            if (strpos($this->request->getHeaderLine('accept'), 'text/html') === false) {
+                $this->respond(ENVIRONMENT === 'development' ? $this->collectVars($this->exception, $this->statusCode) : '', $this->statusCode)->send();
+
+                exit($this->exitCode);
+            }
+        }
+
+        // Determine possible directories of error views
+        $path    = $this->viewPath;
+        $altPath = rtrim((new Paths())->viewDirectory, '\\/ ') . DIRECTORY_SEPARATOR . 'errors' . DIRECTORY_SEPARATOR;
+
+        $path    .= (is_cli() ? 'cli' : 'html') . DIRECTORY_SEPARATOR;
+        $altPath .= (is_cli() ? 'cli' : 'html') . DIRECTORY_SEPARATOR;
```

Comment:
repetitive same assign expr can be moved to a variable before:

```suggestion
        $addPath  = (is_cli() ? 'cli' : 'html') . DIRECTORY_SEPARATOR;
        $path    .= $addPath;
        $altPath .= $addPath;
```